elks/arch/i86/drivers/char/serial.c

```diff
@@ -432,15 +436,19 @@ static void rs_init(void)
 	if (!rs_probe(sp)) {
 	    switch(sp->irq) {
 	    default:
-#if defined(CONFIG_NEED_IRQ4) || defined(CONFIG_NEED_IRQ3)
+#if !defined(CONFIG_FAST_IRQ3) && !defined(CONFIG_FAST_IRQ43)
 		request_irq(sp->irq, rs_irq, NULL);
 #endif
 		break;
 #ifdef CONFIG_FAST_IRQ4
 	    case 4:
+		// FIXME: how to find correct code segment of the fast handler ?
+		int_vector_set (0x0C, _irq_com1, kernel_cs);
```

Comment:
Replace int_vector_set with request_irq(sp->irq, _irq_com1, 1) using changed third parameter as flag indicating no trampoline used. 

If you don't want to pass far pointer to handler now, that's OK, we can simplify things and limit interrupt routines to be in the near kernel code segment for now, which is same limitation as all other handler routines, including fast _irq_com1.

Remove enable_irq on line 453.

Regarding runtime switching between fast and slow serial driver: this first version won't allow that since we don't have way to de- or re-allocate vectors. But no problem, still could have benefit of compiling both in, then having runtime option of selecting one or other at system start, using /bootopts or another method. This will result in big benefit for serial network testing, which is reason for fast driver anyways.

Reply:
Done in 1db6bc5. Moved the fast handlers in the same module as the generic handler to ensure CS is consistent with `int_vector_set` even in medium memory model.